src/api/java/AST.java

```diff
@@ -28,23 +28,16 @@
      * Object comparison.
      * 
      * @param o another AST
-     **/    
+     **/
+    @Override
     public boolean equals(Object o)
     {
-        AST casted = null;
+        if (o == this) return true;
+        if (!(o instanceof AST)) return false;
+        AST casted = (AST) o;
 
-        try
-        {
-            casted = AST.class.cast(o);
-        } catch (ClassCastException e)
-        {
-            return false;
-        }
-
-        return  (this == casted) ||
-                (this != null) &&
-                (casted != null) &&
-                (getContext().nCtx() == casted.getContext().nCtx()) &&
```

Comment:
If class is known at a compile time, there's no difference between the two.
I have been using Java extensively since about 1.5, and TBH I have never even seen the `class.cast` form.
It looks highly un-idiomatic to me, people either use `getClass() == other.getClass()` to verify the exact class match, or `instanceof` if inheritance is allowed.